### PR TITLE
Minor refresh of Go SDK tutorial

### PR DESCRIPTION
**Pull request by sw-yx**
## What was changed:
annotate Go SDK tutorial with filenames and update tctl commands for latest version

## Why?
it was slightly out of date and hard to parse

### CODE REVIEW

File: docs/go-sdk-video-tutorial.md [hunk 1]
```diff
@@ -8,9 +8,94 @@ import { ResponsivePlayer } from '../src/components'
...
 
-Source code:
+
+:::note tctl CLI commands
+
+The video has outdated `tctl` CLI commands from Temporal v0.20. 
+"Task lists" have been renamed to "task queues" and "domains" renamed to "namespaces" [since v0.28](https://docs.temporal.io/docs/cadence-to-temporal/). 
+We update the commands used accordingly below.
+
+:::
```
Comment (sw-yx) [inline docs/go-sdk-video-tutorial.md:18]:
```suggestion
:::note `tctl` CLI commands

The video has outdated `tctl` CLI commands from Temporal v0.20. 
"Task lists" have been renamed to "task queues" and "domains" renamed to "namespaces" [since v0.28](https://docs.temporal.io/docs/cadence-to-temporal/). 
We update the commands used accordingly below.
You can also check the [`tctl` docs](https://docs.temporal.io/docs/tctl).

:::
```

**Review by flossypurse — APPROVED** [review]
